src/pages/index.js

```diff
@@ -16,79 +22,241 @@ const cardListEl = document.querySelector(".cards__list");
+  addCardModalSubmitButton.innerHTML = "Saving...";
+  api
+    .postCard({ name: title, link: url })
+    .then((cardData) => {
+      renderCard(cardData, cardListEl);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardFormElement.reset();
+      addCardModalSubmitButton.classList.add("modal__button_disabled");
+      newCardModal.close();
+      addCardModalSubmitButton.innerHTML = "Create";
+    });
 }
 
-function handleAddCardFormSubmit(cardData) {
-  renderCard({ name: cardData.title, link: cardData.url }, cardListEl);
-  addCardFormElement.reset();
-  newCardModal.close();
+function handleDeleteCardPopup(card) {
+  deleteCardModal.open(card);
+}
+
+function handleDeleteCard(card) {
+  api
+    .deleteCard(card.getCardId())
+    .then(() => {
+      card._element.remove();
+      deleteCardModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleLikeCard(card) {
+  if (card.isLiked()) {
+    api
+      .unlikeCard(card.getCardId())
+      .then(() => {
+        card._likeButton.classList.remove("card__like-button_active");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .likeCard(card.getCardId())
+      .then(() => {
+        card._likeButton.classList.add("card__like-button_active");
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
 }
 
 function showPreviewImage(data) {
   popupWithImage.open(data);
 }
 
-profileEditButton.addEventListener("click", (e) => {
-  const userData = userInfo.getUserInfo();
-  profileTitleInput.value = userData.title;
-  profileDescriptionInput.value = userData.description;
-  editCardModal.open();
-});
+function handleProfileEditSubmit({ title, description }) {
+  profileEditSubmitButton.innerHTML = "Saving...";
+  api
+    .editUserInfo({ title, description })
+    .then(() => {
+      userInfo.setUserInfo({ name: title, about: description });
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editCardModal.close();
+      profileEditSubmitButton.innerHTML = "Save";
+    });
+}
 
-addNewCardButton.addEventListener("click", () => newCardModal.open());
+function handleChangeProfilePictureSubmit({ picture_url }) {
+  changeProfilePictureSubmitButton.innerHTML = "Saving...";
+  api
+    .editUserProfilePicture({ link: picture_url })
+    .then(() => {
+      profileImage.src = picture_url;
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      changeProfilePictureModal.reset();
+      profilePictureModal.close();
+      changeProfilePictureSubmitButton.innerHTML = "Save";
+    });
+}
+
+/* validation */
+
+const editFormValidator = new FormValidator(
+  validationSettings,
+  profileEditForm
+);
+const addFormValidator = new FormValidator(
+  validationSettings,
+  addCardFormElement
+);
+
+const changeProfilePictureFormValidator = new FormValidator(
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```